### DISCUSSION TITLE
✅ Display widget settings conditionally.

### DISCUSSION THREAD

**Discussion by 79ho3ein** [elementor/elementor#21847] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

Hi elementor team.

In widgets, it is very important that widget settings are hidden when not in use. In fact, widgets settings should be displayed conditionally. This makes options that don't work unavailable to mislead users. Especially new users of Elementor.
Conditional display of options has not yet been observed in Elementor. So I want to mention several of them. I hope you agree with me.

1- In the button widget, even if we don't select an icon, the Icon Spacing option is still displayed. **While this option should only be displayed if we select an icon.**

2- In Container > Style tab > Background, if we don't select any background color or image in the hover section, the Transition Duration option will still be displayed. **While this option should be displayed only if we choose the background color or image.**

3- In the nested tabs widget, even if we do not select any icon, there is still an icon customization section in the style tab.

4- In the Mega Menu widget, even if we don't enable the Dropdown Content option, the Dropdown Indicator section is still displayed. 

5- In the Widget Mega Menu > Effect dropdown, the Animation Duration option will not work until we select an Animation. So why should this option be displayed by default? **It is better if the animation is set to none, then the Animation Duration option is hidden.**

6- In the Megamenu widget, **when we enable Dropdown Content**, we will definitely not need the Link option. Therefore, it is better to hide the Link option.

and many other similar cases

### Describe the solution you'd like

Display settings conditionally in old and new widgets. This allows us to only deal with options that are useful and work.

thank you

**Comment by rami-elementor** [comment]
@79ho3ein Thank you for your valuable feedback.

1. [Fixed in v3.20](https://github.com/elementor/elementor/pull/24918)
2. [Planed for 3.26](https://github.com/elementor/elementor/pull/29274)

**Comment by nicholaszein** [comment]
Hey @79ho3ein!

We have great news! 🙌 

### 📢 We're happy to announce that this feature was released with Elementor v3.26.0! 🥳 

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
